conduit/data/datamodules/audio/base.py

```diff
@@ -0,0 +1,89 @@
+"""Base class for audio datasets."""
+from __future__ import annotations
+from abc import abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+import albumentations as A
+from kit import implements
+from kit.torch import TrainingMode
+
+from conduit.data.datamodules.base import CdtDataModule
+from conduit.data.datasets.utils import AlbumentationsTform
+from conduit.data.datasets.wrappers import ImageTransformer, InstanceWeightedDataset
+from conduit.data.structures import ImageSize
+from conduit.types import Stage
+
+__all__ = ["CdtAudioDataModule"]
+
+
+class CdtAudioDataModule(CdtDataModule):
+    _input_size: ImageSize
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        *,
+        train_batch_size: int = 64,
+        eval_batch_size: Optional[int] = 100,
+        num_workers: int = 0,
+        val_prop: float = 0.2,
+        test_prop: float = 0.2,
+        seed: int = 47,
+        persist_workers: bool = False,
+        pin_memory: bool = True,
+        stratified_sampling: bool = False,
+        instance_weighting: bool = False,
+        training_mode: Union[TrainingMode, str] = "epoch",
+    ):
+        super().__init__(
+            train_batch_size=train_batch_size,
+            eval_batch_size=eval_batch_size,
+            num_workers=num_workers,
+            persist_workers=persist_workers,
+            pin_memory=pin_memory,
+            seed=seed,
+            test_prop=test_prop,
+            val_prop=val_prop,
+            stratified_sampling=stratified_sampling,
+            instance_weighting=instance_weighting,
+            training_mode=training_mode,
+        )
+        self.root = root
+
+    @property
+    def size(self) -> ImageSize:
+        if hasattr(self, "_input_size"):
+            return self._input_size
+        if hasattr(self, "_train_data"):
+            self._input_size = ImageSize(*self._train_data[0].x.shape)  # type: ignore
+            return self._input_size
+        raise AttributeError("Input size unavailable because setup has not yet been called.")
+
+    @property
+    @abstractmethod
+    def _base_augmentations(self) -> A.Compose:
+        ...
+
+    @property
+    @abstractmethod
+    def _train_augmentations(self) -> A.Compose:
+        ...
+
+    @implements(CdtDataModule)
+    def setup(self, stage: Stage | None = None) -> None:
+        train, val, test = self._get_splits()
+        train = ImageTransformer(train, transform=self._augmentations(train=True))
+        if self.instance_weighting:
+            train = InstanceWeightedDataset(train)
+        self._train_data = train
+        self._val_data = ImageTransformer(val, transform=self._augmentations(train=False))
+        self._test_data = ImageTransformer(test, transform=self._augmentations(train=False))
```

Comment:
You'll need to adapt the ImageTransformer class into an AudioTransformer class that calls apply_waveform_transform instead of apply_image_transform.

Reply:
Ok will do